### PR TITLE
Leaderboard: feature hit the api

### PR DESCRIPTION
**Pull request by jmoseka**
The following is implemented in this branch

- Send and receive data from an API.
- "Refresh" button (receiving data from the API and parsing the JSON).
- "Submit" button (sending data to the API).
- Used async and await JavaScript features to consume the API.

### CODE REVIEW

File: src/modules/GetList.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+async function getList() {
+  const url = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/Zl4d7IVkemOTTVg2fUpp/scores/';
```
Comment (Cathella) [inline src/modules/GetList.js]: - [ ] Kindly use **arrow functions** as requested in the project requirements

File: src/modules/GetList.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const response = await fetch(url);
+  const reply = await response.json();
+  const scoreListContainer = document.querySelector('.score-list-container');
```
Comment (Cathella) [inline src/modules/GetList.js]: ✅ Used `async` and `await` with the `fetch` method to send and receive data 👍🏻 
✅ When the Refresh button is clicked, it should hit the GET API endpoint causing the scores list to change 👍🏻

File: src/modules/GetList.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  scoreListContainer.innerHTML = '';
+  const sorted = reply.result.sort((a, b) => a.score - b.score);
+  // eslint-disable-next-line no-restricted-syntax
```
Comment (Cathella) [inline src/modules/GetList.js]: - [ ] Kindly enable the linters and instead work on fixing the linter errors 🙏🏻. It helps with keeping consistent code standards

File: src/modules/AddList.js [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
-export default function addList(add) {
-  const list = getStorage();
-  list.push(add);
-  setStorage(list);
+export default async function addList(newData) {
+  const response = await fetch(baseUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      // 'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (Cathella) [inline src/modules/AddList.js]: - [ ] Kindly delete all commented out code. It helps to keep the code-base clean and readable.

File: src/index.js [hunk 5]
```diff
@@ -1,30 +1,38 @@
...
     notice.classList.remove('hidden');
   } else {
-    const name = nameInput.value;
+    const user = nameInput.value;
     const score = scoreInput.value;
-    const scoreClass = new ScoreCL(name, score);
+    const scoreClass = new ScoreCL(user, score);
     addList(scoreClass);
-    displayList();
+    // displayList();
```
Comment (Cathella) [inline src/index.js]: - [ ] Kindly delete or commented out code. It helps to keep the code-base clean and readable.

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @jmoseka 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ Professional README Structure.
✅ Descriptive pull request.
✅ App runs as expected. You can submit user data and it reflects on the scoreboard


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jmoseka** [comment]
✅ Removed unused comments
✅ Replaced normal functions to callback 
✅ Update loop style to avoid usage of ESlint disable lines

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @jmoseka, 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**